src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -25,14 +31,46 @@ public class FindCommand extends Command {
 
     private final Predicate<Person> predicate;
 
+    private Index targetIndex;
+
+    private FindPersonDescriptor findPersonDescriptor;
+
     public FindCommand(Predicate<Person> predicate) {
         this.predicate = predicate;
     }
 
+    /**
+     * Constructor for FindCommand.
+     *
+     * @param findPersonDescriptor contains details for the customer we are finding.
+     */
+    public FindCommand(FindPersonDescriptor findPersonDescriptor) {
+        this.predicate = null;
+        this.findPersonDescriptor = findPersonDescriptor;
+    }
+
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        if (!isNull(predicate)) {
+            model.updateFilteredPersonList(predicate);
+        } else {
+            if (findPersonDescriptor.isPhoneEmpty) {
+                this.targetIndex = Index.fromZeroBased(model.findEmail(findPersonDescriptor.getEmail()));
+            } else {
+                this.targetIndex = Index.fromZeroBased(model.findNum(findPersonDescriptor.getPhone()));
```

Comment:
Maybe we will need to make edits to handle the PersonNotFoundException after updating the findEmail and findNum methods since they both throw PersonNotFoundException. 👍